compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrClassifierStorage.kt

```diff
@@ -145,6 +147,15 @@ class Fir2IrClassifierStorage(
         val origin = regularClass.irOrigin(firProvider)
         val irClass = registerIrClass(regularClass, parent, origin)
         processClassHeader(regularClass, irClass)
+        // TODO: queue-based logic?
+        // Force loading Java methods in Java SAM stub, since FunctionReferenceLowering relies on it.
+        if (converter != null && regularClass.origin == FirDeclarationOrigin.Java) {
+            regularClass.getSamIfAny()?.let { sam ->
```

Comment:
I think this logic should be in `addDeclarationsToExternalClass` and not here. `createIrClass` should just create a class.